crypto/evp/evp_utils.c

```diff
@@ -45,3 +45,32 @@ int evp_do_ciph_ctx_setparams(const EVP_CIPHER *ciph, void *provctx,
         return -1;
     return ciph->ctx_set_params(provctx, params);
 }
+
+int evp_do_md_getparams(const EVP_MD *md, OSSL_PARAM params[])
+{
+    if (md->prov == NULL)
+        return -2;
```

Comment:
I seriously dislike these magic error returns. Either it should be a 0 == fail, 1 == success. Or if it has more potential error codes than that we really ought to have macros for the various values.

Reply:
-2 has been a "unsupported" code for as long as I can remember, specifically used in ctrl types of functions.  Since this function is used to implement ctrl functions *among others*, I thought that the best course of action to not cause breakage all over the place was to replicate the behaviour, and compensate where needed (hence all the `return ok != 0 ? ...` in diverse other places)

Regarding macros for these "magic values", they are all over the place, not just here, so that's a clean up job, which *this PR is not*.
PR welcome (not directed at you specifically, @mattcaswell)!